### PR TITLE
feat: add new resource for console customization

### PR DESCRIPTION
**Pull request by loriepisicchio**
## Issue

https://gravitee.atlassian.net/browse/APIM-2954

## Description

Get console customization. This checks that the license allows to customize the theme (i.e. includes OEM customization feature)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-yrphtjcvnc.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 1]
```diff
@@ -5864,6 +5886,49 @@ components:
                     description: The configuration of the rule
             additionalProperties:
                 type: object
+        ConsoleCustomization:
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:5890]: Some descriptions for the properties added would be nice

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/console/exceptions/CustomizationNotAllowed.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.console.exceptions;
+
+import io.gravitee.apim.core.exception.ValidationDomainException;
+
+public class CustomizationNotAllowed extends ValidationDomainException {
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/console/exceptions/CustomizationNotAllowed.java]: I would suggest making a NotAllowedDomainException that can be mapped in MAPI v2 to a 403 response.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.license.domain_service;
+
+public interface LicenseDomainService {
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java]: Have this be the Domain Service (not interface). And create a `LicenseService` + `LicenseServiceImpl` that acts as a proxy for `NodeLicenseService`.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    public static String OEM_CUSTOMIZATION_FEATURE = "oem-customization";
+    boolean isFeatureEnabled(String feature);
+
+    default boolean isFeatureMissing(String feature) {
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java]: I don't think this is being used by the other files?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java (hunk 3 above)
Comment (loriepisicchio) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java]: Fixed it by adding another exception in the archunit test to allow core to use API from Gravitee node

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java (hunk 4 above)
Comment (loriepisicchio) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java]: you're right => 🗑️

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/console/exceptions/CustomizationNotAllowedException.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.console.exceptions;
+
+import io.gravitee.apim.core.exception.AbstractDomainException;
+
+public class CustomizationNotAllowedException extends AbstractDomainException {
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/console/exceptions/CustomizationNotAllowedException.java]: What about creating a general `NotAllowedException` that would be in this package and be mapped in MAPI v2? That way this specific exception can be moved to a domain folder: `/console/`

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-security/src/main/java/io/gravitee/rest/api/management/v2/security/config/BasicSecurityConfigurerAdapter.java [hunk 6]
```diff
@@ -260,6 +260,8 @@ private HttpSecurity authorizations(HttpSecurity security) throws Exception {
              */
             .requestMatchers(HttpMethod.GET, "/ui/bootstrap")
             .permitAll()
+            .requestMatchers(HttpMethod.GET, "/console/customization")
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-security/src/main/java/io/gravitee/rest/api/management/v2/security/config/BasicSecurityConfigurerAdapter.java]: Change to `/ui/customization`

**Review by jourdiw — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-security/src/main/java/io/gravitee/rest/api/management/v2/security/config/BasicSecurityConfigurerAdapter.java (hunk 6 above)
Comment (loriepisicchio) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-security/src/main/java/io/gravitee/rest/api/management/v2/security/config/BasicSecurityConfigurerAdapter.java]: 🤦

**Review by jourdiw — APPROVED** [review]

**Pull request merged**